rllib/examples/envs/custom_env_render_method.py

```diff
@@ -0,0 +1,136 @@
+# TODO (sven): Move this example script into the new API stack.
+
+import argparse
+import gymnasium as gym
+import numpy as np
+import ray
+from gymnasium.spaces import Box, Discrete
+
+from ray import air, tune
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.env.multi_agent_env import make_multi_agent
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--framework",
+    choices=["tf", "tf2", "torch"],
+    default="torch",
+    help="The DL framework specifier.",
+)
+parser.add_argument("--multi-agent", action="store_true")
+parser.add_argument("--stop-iters", type=int, default=10)
+parser.add_argument("--stop-timesteps", type=int, default=10000)
+parser.add_argument("--stop-reward", type=float, default=9.0)
+
+
+class CustomRenderedEnv(gym.Env):
+    """Example of a custom env, for which you can specify rendering behavior."""
+
+    # Must specify, which render modes are supported by your custom env.
+    # For RLlib to render your env via the "render_env" config key, only
+    # mode="rgb_array" is needed. RLlib will automatically produce a simple
+    # viewer for the returned RGB-images for mode="human", such that you don't
+    # have to provide your own window+render handling.
+    metadata = {
+        "render.modes": ["rgb_array"],
+    }
+
+    def __init__(self, config):
+        self.end_pos = config.get("corridor_length", 10)
+        self.max_steps = config.get("max_steps", 100)
+        self.cur_pos = 0
+        self.steps = 0
+        self.action_space = Discrete(2)
+        self.observation_space = Box(0.0, 999.0, shape=(1,), dtype=np.float32)
+
+    def reset(self, *, seed=None, options=None):
+        self.cur_pos = 0.0
+        self.steps = 0
+        return [self.cur_pos], {}
+
+    def step(self, action):
+        self.steps += 1
+        assert action in [0, 1], action
+        if action == 0 and self.cur_pos > 0:
+            self.cur_pos -= 1.0
+        elif action == 1:
+            self.cur_pos += 1.0
+        truncated = self.steps >= self.max_steps
+        done = self.cur_pos >= self.end_pos or truncated
+        return [self.cur_pos], 10.0 if done else -0.1, done, truncated, {}
+
+    def render(self, mode="rgb"):
```

Comment:
Is there any particular reason why we diverge here from the standard `gymnasium` API? The API does

1. Not receive any parameters anymore, i.e. `def render(self)`
2. Does return type: `RenderFrame | list[RenderFrame] | None` but not `bool`.

Reply:
So sorry, this was a left-over from the next PR. This entire script has been re-done and packed into a separate PR. Removed from this one.